kin/account.py

```diff
@@ -0,0 +1,347 @@
+"""Contains the KinAccount and AccountStatus classes."""
+
+from functools import partial
+
+from enum import Enum
+from stellar_base.asset import Asset
+
+from .blockchain.keypair import Keypair
+from .blockchain.horizon import Horizon
+from .blockchain.builder import Builder
+from .blockchain.channel_manager import ChannelManager
+from . import errors as KinErrors
+from .transactions import Transaction
+from .blockchain.errors import TransactionResultCode, HorizonErrorType , HorizonError
+from .config import MIN_ACCOUNT_BALANCE, SDK_USER_AGENT
+from .blockchain.utils import is_valid_secret_key, is_valid_address
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class KinAccount:
+    """Account class to perform authenticated actions on the blockchain"""
+    def __init__(self, seed, client, channels, channel_secret_keys, create_channels):
+        # Set the internal sdk
+        self._client = client
+
+        # Verify seed
+        if not is_valid_secret_key(seed):
+            raise ValueError('invalid secret key: {}'.format(seed))
+
+        # Set keypair
+        self.keypair = Keypair(seed)
+        # check that sdk wallet account exists and is activated, issuer should continue without activation
+        if self._client.get_account_status(self.keypair.public_address) != AccountStatus.ACTIVATED and \
+                        self.keypair.public_address != self._client.kin_asset.issuer:
+            raise KinErrors.AccountNotActivatedError
+
+        if channels is not None and channel_secret_keys is not None:
+            raise ValueError("Account cannot be initialized with both 'channels'"
+                             " and 'channel_secret_keys' parameters")
+
+        if channel_secret_keys is not None:
+            # Use given channels
+            for channel_key in channel_secret_keys:
+                # Verify channel seed
+                if not is_valid_secret_key(channel_key):
+                    raise ValueError('invalid channel key: {}'.format(channel_key))
+                # Check that channel accounts exists (they do not have to be activated).
+                channel_address = Keypair.address_from_seed(channel_key)
+                if self._client.get_account_data(channel_address) == AccountStatus.NOT_CREATED:
+                    raise KinErrors.AccountNotFoundError
```

Comment:
not clear what account is not found, maybe dedicated channel error with inner error as account not found? (or it's too much Java oriented style...? :man_shrugging:)

Reply:
No you are right, I will add that